rust/opendp_tooling/src/codegen/python.rs

```diff
@@ -481,14 +475,16 @@ fn generate_public_example(func: &Function, type_arg: &Argument) -> Option<Strin
 
 /// the generated code ensures every type arg is a RuntimeType, and constructs derived RuntimeTypes
 fn generate_type_arg_formatter(func: &Function) -> String {
+    let type_names = func.type_names();
     let type_arg_formatter: String = func.args.iter()
         .filter(|arg| arg.is_type)
         .map(|type_arg| {
             let name = type_arg.name.as_ref().expect("type args must be named");
-            let generics = if type_arg.generics.is_empty() {
+            let generics = type_arg.generics(&type_names);
+            let generics = if generics.is_empty() {
```

Comment:
Potential for confusion when the same variable is representing different types on different lines, but just a preference.